iroh/tests/gc.rs

```diff
@@ -0,0 +1,440 @@
+#![cfg(all(feature = "mem-db", feature = "iroh-collection"))]
```

Comment:
What about also running the tests with the `fs` db? Would probably be good to make sure deletions there work similarly good

Reply:
The "gc_flat_stress" and "gc_flat_partial" tests are using the flat file db.

The idea was to use a "perfect" db in the smoke tests to see if the gc_loop / driver is implemented correctly, and then use db specific tests like gc_flat_xxx to make sure that the db actually works.

I guess I can make the tests generic.